src/Microsoft.DotNet.Build.Tasks.Feed/PushToBlobFeed.cs

```diff
@@ -97,5 +197,84 @@ public override bool Execute()
                 }
             }
         }
+
+        private static IEnumerable<PackageArtifactModel> ConcatPackageArtifacts(
+            IEnumerable<PackageArtifactModel> artifacts,
+            IEnumerable<ITaskItem> items)
+        {
+            return artifacts.Concat(items
+                .Select(CreatePackageArtifactModel));
+        }
+
+        private static IEnumerable<BlobArtifactModel> ConcatBlobArtifacts(
+            IEnumerable<BlobArtifactModel> artifacts,
+            IEnumerable<ITaskItem> items)
+        {
+            return artifacts.Concat(items
+                .Select(CreateBlobArtifactModel)
+                .Where(blob => blob != null));
+        }
+
+        private static PackageArtifactModel CreatePackageArtifactModel(ITaskItem item)
+        {
+            NupkgInfo info = new NupkgInfo(item.ItemSpec);
+
+            return new PackageArtifactModel
+            {
+                Attributes = ParseCustomAttributes(item),
+                Id = info.Id,
+                Version = info.Version
+            };
+        }
+
+        private static BlobArtifactModel CreateBlobArtifactModel(ITaskItem item)
+        {
+            string virtualPath = item.GetMetadata("RelativeBlobPath") ?? string.Empty;
+            if (virtualPath.StartsWith(AssetsVirtualDir, StringComparison.Ordinal))
+            {
+                virtualPath = virtualPath.Substring(AssetsVirtualDir.Length);
+            }
+            else
+            {
+                // Don't include assets in the manifest if they aren't in "assets/".
+                return null;
+            }
+
+            string[] virtualPathSegments = virtualPath.Split('/');
```

Comment:
Do you need RemoveEmptyEntries here?

Reply:
Obsolete because I'm removing the tree structure, but no, reason was to be compatible with unnamed virtual dirs (`foo//bar` being `'foo'` `''` `'bar'`) if they happen to be pushed.